Secret-agency-service/src/main/java/cz/fi/muni/pa165/secretagency/service/DepartmentServiceImpl.java

```diff
@@ -0,0 +1,98 @@
+package cz.fi.muni.pa165.secretagency.service;
+
+import cz.fi.muni.pa165.secretagency.dao.DepartmentDao;
+import cz.fi.muni.pa165.secretagency.entity.Department;
+import cz.fi.muni.pa165.secretagency.enums.DepartmentSpecialization;
+import cz.fi.muni.pa165.secretagency.service.exceptions.DepartmentServiceException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Implementation of service access to the {@link Department} entity.
+ * This class is part of the service module of the application
+ * that provides the implementation of the business logic (main logic of the application).
+ *
+ * @author Milos Silhar (433614)
+ */
+@Service
+@Transactional
+public class DepartmentServiceImpl extends GenericServiceImpl<Department, DepartmentDao> implements DepartmentService {
+
+    @Override
+    public List<Department> getDepartmentByCity(String city) {
+        return getDao().getDepartmentsInCity(city);
+    }
+
+    @Override
+    public List<Department> getDepartmentByCountry(String country) {
+        if (country == null) {
+            throw new NullPointerException("country is null");
```

Comment:
Tyhle hlášky bych psal ve tvaru: "Country must be set". To, že je null víš z toho, že vyhazuješ NullPointerException. 😄